drivers/wifi/esp_at/esp.c

```diff
@@ -975,7 +980,8 @@ static int esp_mgmt_connect(const struct device *dev,
 	struct esp_data *data = dev->data;
 	int len;
 
-	if (!net_if_is_up(data->net_iface)) {
+	if (!net_if_is_carrier_ok(data->net_iface) ||
```

Comment:
Please also update `esp_mgmt_scan()` to check for `net_if_is_carrier_ok()`, so that scan works in dormant mode.

Reply:
Done